### PR TITLE
Rewrite exporters

### PR DESCRIPTION
**Pull request by steve-fischer-200**
This is an upgrade to Gene List exporter, and superclass, to talk to our svc, rather than iRODS directly.

Only 98% tested because dev iRODS is giving SSL

Ironically we'll be yanking GeneList exporter from galaxy (it doesn't make sense in that context).  But this can be a template going forward.  Subsequent exporters will be easy to do (once we have handlers on the server side)

### CODE REVIEW

File: Tools/config/config.json.sample [hunk 1]
```diff
@@ -1,5 +1,7 @@
 {
+  "service-url": "https://veupathdb.org/dataset-import/user-datasets"
+  "super-user-token": "sfischer%40upenn.edu-a-very-long-number"
```
Comment (dmgaldi) [inline Tools/config/config.json.sample:3]: Do we have a plan for where to ultimately fetch this from?

File: Tools/lib/python/eupath/EupathExporter.py [hunk 2]
```diff
@@ -6,387 +6,229 @@
...
+            os.chdir(temp_path)
+            print_debug("temp path: " + temp_path)
+            self.prepare_data_files(temp_path)
+            tarball_name = self.create_tarball(temp_path)
+            json_body = self.create_body_for_post()
+            print_debug(json_body)
+            user_dataset_id = self.post_metadata_json(json_body)
+            print_debug("UD ID: " + user_dataset_id)
+            self.post_datafile(user_dataset_id, tarball_name)
+            self.poll_for_upload_complete(user_dataset_id)   # teriminates if system or validation error
```
Comment (dmgaldi) [inline Tools/lib/python/eupath/EupathExporter.py:122]: What's the UX if this throws?

File: Tools/lib/python/eupath/EupathExporter.py [hunk 3]
```diff
@@ -6,387 +6,229 @@
...
-        """
-        os.mkdir(temp_path + "/" + self.DATAFILES)
-        for dataset_file in self.identify_dataset_files():
-            shutil.copy(dataset_file['path'], temp_path + "/" + self.DATAFILES + "/" + self.clean_file_name(dataset_file['name']))
-
-    def create_tarball(self):
+    def create_tarball(self, temp_path):
         """
         Package the tarball - contains meta.json, dataset.json and a datafiles folder containing the
         user's dataset files
```
Comment (dmgaldi) [inline Tools/lib/python/eupath/EupathExporter.py]: Comment out of date now

File: Tools/lib/python/eupath/EupathExporter.py [hunk 4]
```diff
@@ -6,387 +6,229 @@
...
-        Used to insure that the calling ip is the one expected (i.e., the one for which the
-        firewall is opened).  In Globus Dev Galaxy instance calling the tool outside of Galaxy
-        versus inside Galaxy resulted in different calling ip addresses.
-        """
-        request = "http://ifconfig.co"
-        headers = {"Accept": "application/json"}
+
+    def poll_for_upload_complete(self, user_dataset_id):
+        start_time = time.time()
+        poll_interval = 1
```
Comment (dmgaldi) [inline Tools/lib/python/eupath/EupathExporter.py]: Nit: poll_interval_seconds for variable name?

File: Tools/config/config.json.sample (hunk 1 above)
Comment (steve-fischer-200) [reply Tools/config/config.json.sample:3]: the token will be for the super user account.  we can scrape it from a website request

File: Tools/lib/python/eupath/EupathExporter.py (hunk 2 above)
Comment (steve-fischer-200) [reply Tools/lib/python/eupath/EupathExporter.py:122]: in Galaxy, when a processing step fails, the user is shown stdout/stderr

File: Tools/bin/exportGeneListToEuPathDB [hunk 5]
```diff
@@ -1,44 +1,13 @@
 # !/usr/bin/env python
 
 import sys
-#sys.path.insert(0, "/Users/crisl-adm/Tools/galaxy/galaxy/tools/eupath/Tools/lib/python")
+#sys.path.insert(1, "/home/ross/EuPathGalaxy/Tools/lib/python/")
```
Comment (Foxcapades) [inline Tools/bin/exportGeneListToEuPathDB:4]: What is this pointing to?  Since it is commented out, should it be removed?

**Review by Foxcapades — APPROVED** [review]

File: Tools/lib/python/eupath/BigwigFilesEupathExporter.py [hunk 6]
```diff
@@ -1,68 +1,83 @@
...
-            elif not args[i+1].endswith(".bw"):
-                args[i+1] = args[i+1] + ".bw"
+            if filename.endswith(".bigwig"):
+                filename = filename[0:-6] + "bw"
+            elif not filename.endswith(".bw"):
+                filename = filename + ".bw"
 
             # check file size
-            size = os.stat(args[i]).st_size
+            size = os.stat(filename).st_size
```
Comment (Foxcapades) [inline Tools/lib/python/eupath/BigwigFilesEupathExporter.py]: Should this be `filename` or `path`?  If it's supposed to be `filename` wont changing the variable value above break the reference?

File: Tools/lib/python/eupath/RnaSeqEupathExporter.py [hunk 7]
```diff
@@ -5,46 +5,90 @@
...
+            - second tuple is strand two
+
+OUTPUT
+  files given cannonical names:
+     unstranded: sample.suffix (with sample name cleaned of icky characters)
+     stranded:  
+       -SAMPLE_NAME.forward.bw
+       -SAMPLE_NAME.reverse.bw
+       -SAMPLE_NAME.one.txt
+       -SAMPLE_NAME.two.txt
```
Comment (Foxcapades) [inline Tools/lib/python/eupath/RnaSeqEupathExporter.py]: are the `.txt` and `.bw` files in this example reversed, or do they get reversed by the process?

**Review by Foxcapades — APPROVED** [review]

**Review by Foxcapades — APPROVED** [review]

**Pull request merged**